vignettes/cli-config-user.Rmd

```diff
@@ -9,18 +9,83 @@ editor_options:
+
+### `cli.hyperlink_*` options and `R_CLI_HYPERLINK_*` env vars
+
+cli supports a few special types of hyperlink that don't just point to, e.g., a webpage or a file.
+These specialized hyperlinks cause R-specific actions to happen, such executing a bit of R code or opening specific documentation.
+
+Set the relevant option or env var to `true`, `TRUE`, or `True` to tell cli that the terminal is capable of implementing these specialized behaviours.
+Leave this configuration unset (or set to anything else) when there is no support for a specific type of hyperlink.
+
+| Action            | Example inline markup                                                                 | Option                   | Env var                    |
+|-------------------|---------------------------------------------------------------------------------------|--------------------------|----------------------------|
+|        Run R code | `{.run testthat::snapshot_review()}`                                                  | `cli.hyperlink_run`      | `R_CLI_HYPERLINK_RUN`      |
+| Open a help topic | `{.fun stats::lm}` `{.topic tibble::tibble_options}` `{.help [{.fun lm}](stats::lm)}` | `cli.hyperlink_help`     | `R_CLI_HYPERLINK_HELP`     |
+| Open a vignette   | `{.vignette tibble::types}`                                                           | `cli.hyperlink_vignette` | `R_CLI_HYPERLINK_VIGNETTE` |
+
+In all cases, the option takes priority over the corresponding env var.
+
+### `cli.hyperlink_*_url_format` options and `R_CLI_HYPERLINK_*_URL_FORMAT` env vars
+
+Recall the overall structure of cli's hyperlinks:
+
+```
+OSC 8 ; {OPTIONAL PARAMS } ; {URI} ST {LINK TEXT} OSC 8 ; ; ST
+```
+
+The `URI` part has a default format for each type of hyperlink, but it is possible to provide a custom format via an option or an env var.
+If defined, the option takes priority over the env var.
+
+| Action            | Default URI format        | Customize via option                | Customize via env var                 |
+|-------------------|---------------------------|-------------------------------------|---------------------------------------|
+|        Run R code | `x-r-run:{code}`          | `cli.hyperlink_run_url_format`      | `R_CLI_HYPERLINK_RUN_URL_FORMAT`      |
+| Open a help topic | `x-r-help:{topic}`        | `cli.hyperlink_help_url_format`     | `R_CLI_HYPERLINK_HELP_URL_FORMAT`     |
+| Open a vignette   | `x-r-vignette:{vignette}` | `cli.hyperlink_vignette_url_format` | `R_CLI_HYPERLINK_VIGNETTE_URL_FORMAT` |
+
+A format must be a glue-like template with the relevant placeholder in curly braces (`code`, `topic` or `vignette`).
+
+Here's an example of a custom URI format for runnable code, which is useful in an integrated Positron terminal:
+
+```
+positron://positron.positron-r/cli?command=x-r-run:{code}
+```
+
+(For backwards compatibility with older versions of RStudio, in some contexts, a legacy format is used, e.g. `ide:run:{code}`.)
```

Comment:
Do I have this right?

Reply:
I am not sure what you mean. Don't we always use `ide::run` in RStudio?

Reply:
> Don't we always use ide::run in RStudio?

I am finding it hard to actually figure this out. The RStudio detection logic is pretty complicated.

I thought there was some plan to start using the `x-r-*` in RStudio as well and to phase out the legacy format.